### ISSUE TITLE
UndefinedVariable & UndefinedGlobalVariable in argument 2 of in_array causes erroneous error in argument 3

### ISSUE THREAD

**Issue by SignpostMarv** [vimeo/psalm#1596]
If argument 2 passed to `in_array()` is undefined, argument 3 is flagged as mixed despite being a literal bool.

* using ae53150
* https://psalm.dev/r/101c5ad5c8

```php
<?php

$a = 0;

echo in_array($a, $b, true) ? 'yes' : 'no';

function Foo(int $a) : bool {
  return in_array($a, $b, true);
}
```

> Psalm output (using commit ae53150): 
> 
> ERROR: UndefinedGlobalVariable - 5:19 - Cannot find referenced variable $b in global scope
> 
> INFO: MixedArgument - 5:19 - Argument 2 of in_array cannot be mixed, expecting array<array-key, mixed>
> 
> INFO: MixedArgument - 5:23 - Argument 3 of in_array cannot be mixed, expecting bool
> 
> ERROR: UndefinedVariable - 8:23 - Cannot find referenced variable $b
> 
> INFO: MixedArgument - 8:23 - Argument 2 of in_array cannot be mixed, expecting array<array-key, mixed>
> 
> INFO: MixedArgument - 8:27 - Argument 3 of in_array cannot be mixed, expecting bool

**Comment by weirdan** [comment]
Can't reproduce anymore.
